Validation/HGCalValidation/plugins/HGCalValidator.cc

```diff
@@ -146,7 +147,7 @@ void HGCalValidator::cpParametersAndSelection(const Histograms& histograms,
   for (auto const caloParticle : cPeff) {
     int id = caloParticle.pdgId();
 
-    if (cpSelector(caloParticle, simVertices)) {
+    if (!doCaloParticleSelection_ || (doCaloParticleSelection_ && cpSelector(caloParticle, simVertices))) {
```

Comment:
Isn't better a simpler `     if (!doCaloParticleSelection_ || cpSelector(caloParticle, simVertices)) {` ?

Reply:
I guess the compiler will optimize it this way.
Can I propose to couple this with a coming PR from Leonardo that will restructure a great portion of this code/class?

Reply:
ok, no problem